packages/vite/src/node/utils.ts

```diff
@@ -1035,9 +1037,60 @@ export async function resolveServerUrls(
         }
       })
   }
+
+  const cert =
+    httpsOptions?.cert && !Array.isArray(httpsOptions.cert)
+      ? new crypto.X509Certificate(httpsOptions.cert)
+      : undefined
+  const hostnameFromCert = cert?.subjectAltName
+    ? extractHostnamesFromSubjectAltName(cert.subjectAltName)
+    : []
+
+  if (hostnameFromCert.length > 0) {
+    const existings = new Set([...local, ...network])
+    local.push(
```

Comment:
I went with adding all of these to `local` so that `Network: use --host to expose` is shown when `--host` is not passed.
We can also hide hosts that resolves to addresses that are not loopback addresses when `--host` is not passed, but I felt that is confusing.